torch/csrc/api/include/torch/nn/functional/upsampling.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <torch/nn/functional/pooling.h>
+#include <torch/nn/options/upsampling.h>
+
+#include <cmath>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor interpolate(const Tensor& input, InterpolateOptions options) {
```

Comment:
I believe the Python version can take no options as well, and in C++ we might want
```suggestion
inline Tensor interpolate(const Tensor& input, InterpolateOptions options = {}) {
```
It would be awesome to add a test for `F::interpolate(input)` (i.e. no options) use case as well :D

Reply:
I don't believe we'll be able to do this because the default `InterpolateOptions` is not valid. Specifically, `size` and `scale_factor` both default to empty vectors but one of these vectors must be defined to compute `_output_size()`. See lines 14-15 of this file in the `_check_size_scale_factor` lambda.

Reply:
Ah that makes sense, thanks a lot for the catch!